Formplayer/testPages/webapps/webapps_basics.py

```diff
@@ -0,0 +1,212 @@
+        self.question_display_text = (By.XPATH, "//span[text()='Name (es)']")
+
+        #login xpaths
+        self.username_textbox_id = (By.ID, "id_auth-username")
+        self.password_textbox_id = (By.ID, "id_auth-password")
+        self.submit_button_xpath = (By.XPATH, '(//button[@type="submit"])[last()]')
+
+        self.reports_menu_id = (By.LINK_TEXT, "Reports")
+        self.submit_history_rep = (By.LINK_TEXT, "Submit History")
+        self.case_list_rep = (By.LINK_TEXT, "Case List")
+        self.apply_id = (By.ID, "apply-filters")
+
+        # Submit History
+        self.users_box = (By.XPATH, "//span[@class='select2-selection select2-selection--multiple']")
+        self.select_user = (By.XPATH, "//li[contains(text(),'[All Data]')]")
+        self.application_select = (By.XPATH, "//select[@id='report_filter_form_app_id']")
+        self.module_select = (By.XPATH, "//select[@id='report_filter_form_module']")
+        self.form_select = (By.XPATH, "//select[@id='report_filter_form_xmlns']")
+        self.case_type_select = (By.XPATH, "//select[@id='report_filter_case_type']")
+        self.date_input = (By.XPATH, "//input[@id='filter_range']")
+        self.view_form_link = (By.XPATH, "//tbody/tr[1]/td[1]/a[.='View Form']")
+        # self.case_name = (By.XPATH, "//td[div[contains(text(),'abc')]]")
+        self.submit_history_table = (By.XPATH, "//table[@id='report_table_submit_history']/tbody/tr")
+
+        # Case List
+        self.search_input = (By.XPATH, "//input[@id='report_filter_search_query']")
+        self.case_list_table = (By.XPATH, "//table[@id='report_table_case_list']/tbody/tr")
+        self.case_id_block = (By.XPATH, "//th[@title='_id']/following-sibling::td")
+
+        self.click_outside = (By.XPATH, "//label[.//span[text()='Mobile No.']]")
+
+    def application_is_present(self):
+        self.wait_to_click(self.webapps_menu_id)
+        assert self.is_present_and_displayed(self.test_application), "Application is not present in Web Apps"
+        print("App is present")
+
+    def login_as_a_user(self):
+        self.wait_to_click(self.login_as_button)
+        self.wait_to_clear_and_send_keys(self.filter_workers, "appiumtest")
+        self.wait_to_click(self.search_button)
+        self.wait_to_click(self.login_user)
+        self.wait_to_click(self.confirm_login_button)
+
+    def submit_form(self):
+        self.wait_to_click(self.webapps_menu_id)
+        self.login_as_a_user()
+        self.wait_to_click(self.test_application)
+        self.wait_to_click(self.case_list_menu)
+        self.wait_to_click(self.registration_form)
+        self.wait_to_clear_and_send_keys(self.name_question, self.name_input)
+        self.wait_to_click(self.dob_question)
+        self.wait_to_click(self.click_today_date)
+        self.wait_to_click(self.close_date_picker)
+        self.wait_to_clear_and_send_keys(self.mobileno_question, UserData.mobile_number + Keys.TAB)
```

Comment:
call the mobile number generating function here. Or is there any specific reason for saving it as a user data?